src/client/components/Event.js

```diff
@@ -14,7 +14,7 @@ function eventTab(selectedTab, eventId) {
     case 'overview':
       return <DataLoader serviceCall={`events/${eventId}`} component={EventOverview} />;
     case 'teams':
-      return <EventTeams />;
+      return <DataLoader serviceCall={`events/${eventId}`} component={EventDepartments} />;
```

Comment:
So... this is part of the React component magic?

Reply:
Yup. `DataLoader` is a component I wrote to handle loading and caching data from the server. You give it a service call path and a component and it hands the response as props to the provided component. If it's not already cached, this triggers an AJAX request and displays a loading spinner instead of the provided component until the request completes.